src/glum/_glm.py

```diff
@@ -1552,6 +1616,53 @@ def score(
         dev_null = family.deviance(y, y_mean, sample_weight=sample_weight)
         return 1.0 - dev / dev_null
 
+    def _compute_information_criteria(
+        self,
+        X: ShapedArrayLike,
+        y: ShapedArrayLike,
+        sample_weight: Optional[ArrayLike] = None,
+    ):
+        """
+        Computes and stores the model's degrees of freedom, and the 'aic',
+        'aicc' and 'bic' information criterion for the model.
+
+        References
+        ----------
+        [1] Burnham KP, Anderson KR (2002). Model Selection and Multimodel
+        Inference; Springer New York.
+        """
+
+        # we require that the log_likelihood be definied
+        if not isinstance(
+            self.family_instance, (BinomialDistribution, TweedieDistribution)
+        ):
+            return False
+
+        if isinstance(
+            self.family_instance, TweedieDistribution
+        ) and self.family_instance.power not in [0, 1, 2]:
+            return False
+
+        ddof = np.sum(np.abs(self.coef_) > np.finfo(self.coef_.dtype).eps)
+
+        k_params = ddof + self.fit_intercept
+
+        mu = self.predict(X)
+        (nobs,) = mu.shape
```

Comment:
```suggestion
        nobs = mu.shape[0]
```
This will work no matter how many dimensions mu has.
It might work right now but I would like to support multi-class models in the near future.

Reply:
As a curiosity: this would also work: `nobs, *_ = mu.shape`.

Reply:
A more interesting point: If the weights represent frequency weights, `nobs` should be the sum of weights instead of the number of rows. If they represent probability weights, the notion of degrees of freedom becomes more complicated.